### PR TITLE
200 update post

### LINKED ISSUE #200: [200] Update Post
**Current state**
Posts cannot be updated

**Expected state**
When is the task done?

**TO DO**
- [x] Detect if profile is owner
- [x] Send changes to database and edit post
- [x] Edit TITLE
- [x] Edit CONTENT

**Necessities**
#185 Posts create and read

**Branch**
200-update-post

### PR DESCRIPTION
**Pull request by AjobK**
closes #200 

 **What is it supposed to do**
With this feature the user is able to update an existing post. Also removed like from posts of owner and added reference to profile two ways (From post to profile and other way around)

 **How can we test your branch**
Explain how can test your branch Ex. width, height etc...

 **What broweser have been checked**
- [x] Chrome
- [x] Firefox
- [x] Opera
- [ ] Edge
- [ ] Safari

### CODE REVIEW

File: backend/app/service/postService.ts [hunk 1]
```diff
@@ -120,21 +120,25 @@ class PostService {
...
 
         const decodedToken = jwt.verify(req.cookies.token, JWT_SECRET)
-        if (post.profile.display_name != decodedToken.username) return res.status(405).json({ 'error': 'Not allowed' })
 
-        if (!post) return res.status(404).json({ 'message': 'post not found' })
+        if (post.profile.display_name != decodedToken.username)
+            return res.status(405).json({ 'error': 'Not allowed' })
+
+        if (!post)
+            return res.status(404).json({ 'message': 'Post not found' })
```
Comment (ryankroon00) [inline backend/app/service/postService.ts:135]: waarom op de volgende lijn?

File: backend/app/service/postService.ts [hunk 2]
```diff
@@ -120,21 +120,25 @@ class PostService {
...
 
         const updatedPost = await this.dao.updatePost(post)
 
-        if (!updatedPost) return res.status(404).json({ 'message': 'post not updated' })
+        if (!updatedPost) return res.status(404).json({ 'message': 'Could not update post' })
```
Comment (ryankroon00) [inline backend/app/service/postService.ts]: waarom niet op de volgende lijn

**Review by ryankroon00 — CHANGES_REQUESTED** [review]
chop chop

File: backend/app/service/postService.ts (hunk 1 above)
Comment (AjobK) [reply backend/app/service/postService.ts:135]: Leesbaarder. Begone

**Comment by jerohero** [comment]
image
Updating twice causes chaos

File: frontend/src/pages/post/index.js [hunk 3]
```diff
@@ -109,7 +120,7 @@ class Post extends App {
         if (!this.props.new) this.loadArticle()
```
Comment (ryankroon00) [inline frontend/src/pages/post/index.js:120]: waarom is dit op dezelfde lijn?????????

**Review by ryankroon00 — CHANGES_REQUESTED** [review]
snel fixen aub

**Review by jerohero — APPROVED** [review]

File: backend/app/controllers/postController.ts [hunk 4]
```diff
@@ -15,9 +15,12 @@ class PostController implements ControllerBase{
 
     public initRoutes(): void {
         this.router.get(this.post, this.postService.getPosts)
-        this.router.get(this.post + '/owned-by/:username', this.postService.getOwnedPosts)
-        this.router.get(this.post + '/:path', this.postService.getPostByPath)
+        this.router.get(this.post + '/:path', this.postService.getPostByPath) 
+        this.router.put(this.post + '/:path', this.postService.updatePost)
```
Comment (ryankroon00) [inline backend/app/controllers/postController.ts]: moet hier geen authenticatie middleware tussen

**Review by ryankroon00 — CHANGES_REQUESTED** [review]

**Review by ryankroon00 — APPROVED** [review]

**Pull request merged**